src/Files.App/UserControls/SidebarControl.xaml.cs

```diff
@@ -1136,32 +1038,31 @@ private void ResizeElementBorder_DoubleTapped(object sender, DoubleTappedRoutedE
 
 		private void ResizeElementBorder_ManipulationStarted(object sender, ManipulationStartedRoutedEventArgs e)
 		{
-			if (DisplayMode == NavigationViewDisplayMode.Expanded)
-			{
-				originalSize = IsPaneOpen ? UserSettingsService.AppearanceSettingsService.SidebarWidth : CompactPaneLength;
-				(sender as Grid).ChangeCursor(InputSystemCursor.Create(InputSystemCursorShape.SizeWestEast));
-				VisualStateManager.GoToState((sender as Grid).FindAscendant<SplitView>(), "ResizerPressed", true);
-				dragging = true;
-			}
+			if (DisplayMode != NavigationViewDisplayMode.Expanded)
+				return;
+
+			originalSize = IsPaneOpen ? UserSettingsService.AppearanceSettingsService.SidebarWidth : CompactPaneLength;
+			(sender as Grid)?.ChangeCursor(InputSystemCursor.Create(InputSystemCursorShape.SizeWestEast));
+			VisualStateManager.GoToState((sender as Grid)?.FindAscendant<SplitView>(), "ResizerPressed", true);
+			dragging = true;
 		}
 
 		private async Task<bool> CheckEmptyDrive(string drivePath)
 		{
-			if (drivePath is not null)
-			{
-				var matchingDrive = App.DrivesManager.Drives.FirstOrDefault(x => drivePath.StartsWith(x.Path, StringComparison.Ordinal));
-				if (matchingDrive is not null && matchingDrive.Type == DriveType.CDRom && matchingDrive.MaxSpace == ByteSizeLib.ByteSize.FromBytes(0))
-				{
-					bool ejectButton = await DialogDisplayHelper.ShowDialogAsync("InsertDiscDialog/Title".GetLocalizedResource(), string.Format("InsertDiscDialog/Text".GetLocalizedResource(), matchingDrive.Path), "InsertDiscDialog/OpenDriveButton".GetLocalizedResource(), "Close".GetLocalizedResource());
-					if (ejectButton)
-					{
-						var result = await DriveHelpers.EjectDeviceAsync(matchingDrive.Path);
-						await UIHelpers.ShowDeviceEjectResultAsync(result);
-					}
-					return true;
-				}
-			}
-			return false;
+			if (drivePath is null)
+				return false;
+
+			var matchingDrive = App.DrivesManager.Drives.FirstOrDefault(x => drivePath.StartsWith(x.Path, StringComparison.Ordinal));
+			if (matchingDrive is null || matchingDrive.Type != DriveType.CDRom || matchingDrive.MaxSpace != ByteSizeLib.ByteSize.FromBytes(0)) 
+				return false;
+
+			var ejectButton = await DialogDisplayHelper.ShowDialogAsync("InsertDiscDialog/Title".GetLocalizedResource(), string.Format("InsertDiscDialog/Text".GetLocalizedResource(), matchingDrive.Path), "InsertDiscDialog/OpenDriveButton".GetLocalizedResource(), "Close".GetLocalizedResource());
+			if (!ejectButton)
```

Comment:
I suggest this is still better because it's not exactly clear why return true if not ejectButton.
Instead, ejectButton just means perform some side effects.
```cs
if (ejectButton) 
{
   ...;
}
return true;
```

Reply:
I'll take the suggestion here, thanks!